plugins/woocommerce/src/Internal/Admin/BlockTemplates/AbstractBlockTemplate.php

```diff
@@ -78,6 +78,20 @@ public function cache_block( BlockInterface &$block ) {
 		$this->block_cache[ $id ] = $block;
 	}
 
+	/**
+	 * Uncaches a block in the template. This is an internal method and should not be called directly
+	 * except for from the BlockContainerTrait's remove_block() method.
+	 *
+	 * @param string $block_id The block ID.
+	 *
+	 * @ignore
+	 */
+	public function uncache_block( string $block_id ) {
```

Comment:
If it is internal should it be `private` or `protected`?

Reply:
I originally had it as `protected`, but that didn't work as it needs to be called from `BlockContainerTrait`. Same reason as to why `cache_block` is public.

I wish there was a better way to handle these sort of "internal" methods in PHP.

Reply:
Gotcha, thanks for the additional explanation. This is fine with me then.